_includes/exhibits.js

```diff
@@ -0,0 +1,1929 @@
+    }, true);
+  },
+
+  newExhibit: function() {
+    const exhibit = this.exhibit;
+    const cgs = deepCopy(exhibit.Groups || []);
+    const stories = deepCopy(exhibit.Stories || []);
+    this.design = {
+      chans: deepCopy(exhibit.Channels || []),
+      layout: deepCopy(exhibit.Layout || {}),
+      images: deepCopy(exhibit.Images || []),
+      stories: stories,
+      cgs: cgs
+    };
+  },
+  newTag: function() {
+    const exhibit = this.exhibit;
+    const stories = deepCopy(exhibit.Stories);
+    const group = this.group;
+    const o = this.o;
+    const v = this.v;
+    const d = this.d;
+    this.stories = [{
+      Description: '',
+      Name: 'Tag',
+      Waypoints: [{
+        Zoom: v[0],
+        Pan: v.slice(1),
+        Group: group.Name,
+        Description: decode(d),
+        Name: 'Tag',
+        Overlay: {
+          x: o[0],
+          y: o[1],
+          width: o[2],
+          height: o[3],
+        },
+      }]
+    }].concat(stories);
+  },
+  pushState: function() {
+    const hashKeys = this.hashable.edits;
+    const searchKeys = this.searchKeys;
+    const url = this.makeUrl(hashKeys, searchKeys);
+    const title = document.title;
+    const design = this.design;
+
+    if (this.url == url && !this.changed) {
+      return;
+    }
+
+    if (!this.embedded && this.hashKeys === hashKeys) {
+      history.pushState(design, title, url);
+    }
+    else {
+      // Replace any invalid state
+      history.replaceState(design, title, url);
+    }
+    window.onpopstate();
+    this.changed = false;
+  },
+  popState: function(e) {
+    if (e && e.state) {
+      this.changed = false;
+      this.design = e.state;
+    }
+    const hash = this.hash;
+    const search = this.search;
+    const hashable = this.hashable;
+    const hashKeys = this.hashKeys;
+    const searchKeys = this.searchKeys;
+
+    // Take search parameters
+    searchKeys.forEach(function(key) {
+      this[key] = search[key];
+    }, this);
+
+    // Accept valid hash
+    hashKeys.forEach(function(key) {
+      this[key] = hash[key];
+    }, this);
+
+    // Setup if invalid hash
+    if (!this.isHash) {
+      this.newExhibit();
+      this.s = 0;
+      this.g = 0;
+      this.pushState();
+    }
+
+    // Do not persist tag in URL
+    if (hashKeys === hashable.tag) {
+      this.newTag();
+      this.s = 0;
+      this.g = 0;
+      this.pushState();
+      $('.modal').modal('hide');
+    }
+
+    // Always update
+    this.newView(true);
+  },
+  newView: function(redraw) {
+
+    // Temp overlay if drawing or editing
+    if (this.drawing || this.editing) {
+      this.addOverlay(this.overlay);
+    }
+    else {
+      this.addOverlay(this.waypoint.Overlay);
+    }
+
+    // Redraw design
+    if(redraw) {
+      // Update OpenSeadragon
+      this.activateViewport();
+      newMarkers(this.tileSources, this.group);
+      // Redraw HTML Menus
+      this.addChannelLegends();
+      this.addGroups();
+      this.newStories();
```

Comment:
This redraws the HTML to match the HashState's current configuration of channel groups and stories.